docs/guides/structured-output.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: default
+title: Structured Output
+parent: Guides
+nav_order: 7
+---
+
+# Structured Output
+
+RubyLLM allows you to request structured data from language models by providing a JSON schema. When you use the `with_response_format` method, RubyLLM will ensure the model returns data matching your schema instead of free-form text.
+
+## Schema-Based Output (Recommended)
+
+We recommend providing a schema for structured data:
+
+```ruby
+# Define a JSON schema
+schema = {
+  type: "object",
+  properties: {
+    name: { type: "string" },
+    age: { type: "integer" },
+    interests: { type: "array", items: { type: "string" } }
+  },
+  required: ["name", "age", "interests"]
+}
+
+response = RubyLLM.chat(model: "gpt-4o")
+  .with_response_format(schema)
+  .ask("Create a profile for a Ruby developer")
+```
+
+RubyLLM intelligently handles your schema based on the model's capabilities:
+
+- For models with native schema support (like GPT-4o): Uses API-level schema validation
+- For models without schema support: Automatically adds schema instructions to the system message
+
+## Simple JSON Mode (Alternative)
+
+For cases where you just need well-formed JSON:
+
+```ruby
+response = RubyLLM.chat(model: "gpt-4.1-nano")
+  .with_response_format(:json)
+  .ask("Create a profile for a Ruby developer")
+```
+
+This uses OpenAI's `response_format: {type: "json_object"}` parameter, works with most OpenAI models, and guarantees valid JSON without enforcing a specific structure.
+
+## Strict and Non-Strict Modes
```

Comment:
I'm not convinced about the naming here. For example for skipping the model registry check I used `assume_model_exists` which clarifies the intention of the flag. The first time I read `strict` I had no idea what the API was gonna be strict about? The output format? How?

I would go for `assume_supported: true`